src/wazuh.py

```diff
@@ -60,12 +86,45 @@ def update_configuration(container: ops.Container, indexer_ips: list[str]) -> No
         new_host = etree.Element("host")
         new_host.text = f"https://{ip_port}"
         hosts[0].append(new_host)
+
+    cluster = ossec_config_tree.xpath("/root/ossec_config/cluster")
+    if cluster:
+        cluster[0].getparent().remove(cluster[0])
+    node_name = unit_name.replace("/", "-")
+    node_type = NodeType.MASTER if unit_name.split("/")[1] == "0" else NodeType.WORKER
     elements = ossec_config_tree.xpath("//ossec_config")
-    content = b""
-    for element in elements:
-        content = content + etree.tostring(element, pretty_print=True)
+    if len(charm_addresses) > 1:
+        new_cluster = etree.fromstring(  # nosec
+            _generate_cluster_snippet(node_name, node_type, charm_addresses, key)
+        )
+        elements[0].append(new_cluster)
+
+    content = b"".join([etree.tostring(element, pretty_print=True) for element in elements])
     container.push(OSSEC_CONF_PATH, content, encoding="utf-8")
 
+
+def update_configuration(
+    container: ops.Container,
+    indexer_ips: list[str],
+    charm_addresses: list[str],
+    unit_name: str,
+    key: str,
+) -> None:
+    """Update the charm configuration.
```

Comment:
Is it really the charm configuration ? I think it's "Wazuh service configuration"

Reply:
Let's rename it as "workload"